rbx_binary/src/deserializer/state.rs

```diff
@@ -399,7 +399,7 @@ impl<'db, R: Read> DeserializerState<'db, R> {
                 VariantType::String => {
                     for referent in &type_info.referents {
                         let instance = self.instances_by_ref.get_mut(referent).unwrap();
-                        let value = chunk.read_string()?;
+                        let value = String::from_utf8_lossy(chunk.read_binary_string()?.as_ref()).to_string();
                         add_property(instance, &property, value.into());
```

Comment:
```suggestion
                        let binary_string = chunk.read_binary_string()?;
                        let value = match std::str::from_utf8(&binary_string) {
                            Ok(value) => Cow::Borrowed(value),
                            Err(_) => {
                                log::warn!(
                            "Performing lossy string conversion on property {}.{} because it did not contain UTF-8.
This may cause unexpected or broken behavior in your final results if you rely on this property being non UTF-8.",
                                    type_info.type_name,
                                    property.name
                                );

                                String::from_utf8_lossy(&binary_string)
                            }
                        };

                        add_property(instance, &property, value.as_ref().into());
```